lua/caramba/intelligence.lua

```
@@ -540,6 +540,123 @@ M._ext_to_filetype = function(ext)
+  
+  -- Check for potential dependencies
+  local deps = M.db.dependencies[file_path] or {}
+  if #deps > 0 then
+    table.insert(context_parts, "\nKnown dependencies:")
+    for _, dep in ipairs(deps) do
+      table.insert(context_parts, "  - " .. dep)
+      
+      -- Check if any symbols might come from this dependency
+      for symbol, _ in pairs(symbols) do
+        -- Simple heuristic: check if symbol name appears in dependency usage
+        local usage_in_dep = vim.fn.system(string.format(
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'vim'

(W113)